src/components/starlight/MarkdownContent.astro

```diff
@@ -0,0 +1,20 @@
+---
+import type { Props } from '@astrojs/starlight/props';
+import '@astrojs/starlight/style/markdown.css';
+
+import { Aside } from '@astrojs/starlight/components';
+import { coerce, satisfies } from 'semver'
+
+const { astroRange } = Astro.props.entry.data;
+---
+
+{
+  // only show if the range is specified, and is not satisfied by the latest version
+  astroRange && !satisfies(Astro.locals.astroLatest, astroRange) && (
+    <Aside type="caution">
+      This guide was written for Astro {coerce(astroRange)} and may not work with more recent versions of Astro.
+    </Aside>
+  )
+}
+
+<div class="sl-markdown-content"><slot /></div>
```

Comment:
Is it possible to wrap the default starlight component instead?

Reply:
Yep!